src/main/java/seedu/duke/account/Account.java

```diff
@@ -0,0 +1,297 @@
+package seedu.duke.account;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import seedu.duke.*;
+import seedu.duke.exception.FinanceException;
+import seedu.duke.exception.FinanceException.ExceptionCollection;
+
+public class Account {
+    protected Wallet wallet;
+    protected String loginTime;
+
+    public Account(Wallet wallet) {
+        this.wallet = wallet;
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("uuuu-MM-dd>HH:mm:ss");
+        loginTime = dtf.format(LocalDateTime.now());
+    }
+
+    public boolean handleAccountRequest() throws FinanceException {
+        boolean isAccountExit = false;
+        AccountUi.showAccountEntryMessage(wallet.getUserName());
+
+        AccountHistoryFile.createNewLoginAccount(wallet.getUserName(), loginTime);
+
+        CurrencyStructure currency;
+        while (!isAccountExit) {
+            String in = InputManager.receiveInputLine().toLowerCase();
+            String[] splits = in.split(" ");
+
+            if (splits.length == 1) {
+                String commandType = splits[0];
+                try {
+                    switch (commandType) {
+                    case "balance":
```

Comment:
The command type could be implemented as an enumeration since it's a list of finite values. It could help more with coding, debugging and reasoning the program.